rice-middleware/kim/kim-impl/src/main/java/org/kuali/rice/kim/impl/group/GroupServiceImpl.java

```diff
@@ -92,6 +92,36 @@ public Group getGroup(String groupId) throws RiceIllegalArgumentException {
     }
 
     @Override
+    public List<Group> getParentGroupsByGroupsIds(List<String> groupIds) {
+        return getParentGroupsByGroupsIds(groupIds, Collections.EMPTY_SET);
+    }
+
+
+    protected List<Group> getParentGroupsByGroupsIds(List<String> groupIds, Set<Group> existingGroups) {
+        final QueryByCriteria.Builder builder = QueryByCriteria.Builder.create();
+        builder.setPredicates(
+                and(
+                        in(KIMPropertyConstants.GroupMember.MEMBER_ID, groupIds),
+                        equal(KIMPropertyConstants.GroupMember.MEMBER_TYPE_CODE, KimConstants.KimGroupMemberTypes.GROUP_MEMBER_TYPE.getCode()),
+                        HistoryQueryUtils.between(KIMPropertyConstants.KimMember.ACTIVE_FROM_DATE_VALUE,
+                                KIMPropertyConstants.KimMember.ACTIVE_TO_DATE_VALUE, new DateTime(System.currentTimeMillis()))));
+
+        Set<Group> parentGroups = new HashSet<Group>(findGroups(builder.build()).getResults());
```

Comment:
This should call `findGroupMembers` first since the `Group` class doesn't have `memberId`, `memberTypeCode and an active date range.